index.js

```diff
@@ -308,6 +309,26 @@ function start(clusterProcessId = 0) { // eslint-disable-line max-statements
   // create app routes
   require('./view/loadView').loadView(app, passportModule, '/v1');
 
+  if (featureToggles.isFeatureEnabled('enableSigtermEvent')) {
+    /*
+     After receiving SIGTERM Heroku will give 30 seconds to shutdown cleanly.
+     If any processes remain after that time period, Dyno manager will terminate
+     them forcefully with SIGKILL logging 'Error R12' to indicate that the
+     shutdown process is not behaving correctly.
+     Steps:
+     - Stop accepting new requests;
+     - Handling pending resources;
+     - If not receive any SIGKILL a timeout will be applied killing the app
+     avoiding zombie process.
+    */
+    process.on('SIGTERM', () => {
+      httpServer.close(() => {
```

Comment:
I found myself needing to check the nodejs api to understand what this close fn was doing--I think it would be useful to at least link to https://nodejs.org/docs/latest-v8.x/api/net.html#net_server_close_callback here when calling the close fn, OR even quote it here

Reply:
done